packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/auth_state_machine.dart

```diff
@@ -55,6 +58,20 @@ class AuthStateMachine extends AuthStateMachineBase {
           endpoint: userPoolConfig.endpoint,
         ),
       );
+
+      // Listen for Analytics Pinpoint ID
+      final analyticsConfig = event.config.analytics;
+      if (analyticsConfig != null) {
+        late StreamSubscription<AnalyticsHubEvent> subscription;
+        subscription = Amplify.Hub.listen(HubChannel.Analytics,
```

Comment:
What happens if configuration fails?

Reply:
Which configuration?

If auth config fails - even if it ran the results are irrelevant as I assume remaining auth will not run. 

If analytics config fails - analytics hub event might not even get here but if it does, that should be okay because => 

Current iOS behavior is for a Pinpoint ID to be generated in iOS Auth even if Analytics is not configured so it appears there was previous consensus that sending non initialized Pinpoint Endpoint ids is okay.  

I'm clarifying this - waiting for Pinpoint team to respond.